msteams-platform/toolkit/use-CICD-template.md

```diff
@@ -10,233 +10,359 @@ ms.date: 04/20/2022
+            uses: actions/checkout@master
+    
+          - name: Setup Node 20.x
+            uses: actions/setup-node@v1
+            with:
+              node-version: "20.x"
+    
+          - name: install cli
+            run: |
+              npm install @microsoft/teamsapp-cli@${{env.TEAMSAPP_CLI_VERSION}}
+    
+          - name: Login Azure by service principal
+            run: |
+              npx teamsapp auth login azure --username ${{vars.AZURE_SERVICE_PRINCIPAL_CLIENT_ID}}  \
+              --service-principal true \
+              --tenant ${{vars.AZURE_TENANT_ID}} \
+              --password ${{secrets.AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET }} \
+              --interactive false
+    
+          - name: Deploy to hosting environment
+            run: |
+              npx teamsapp deploy --ignore-env-file true \
+              --interactive false
+    
+          - name: Package app
+            run: |
+              npx teamsapp package
+    
+          - name: upload appPackage
+            uses: actions/upload-artifact@v4
+            with:
+              name: artifact
+              path: appPackage/build/appPackage.zip
+    ```
+
+    > [!NOTE]
+    > The default pipeline triggers when push events occur on the main branch. You've the option to modify it to suit your specific requirements.
+
+1. Go to GitHub.
+
+1. Locate the following variables and secrets:
+
+    * `AZURE_SERVICE_PRINCIPAL_CLIENT_ID`, `AZURE_TENANT_ID`, and `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET`. [Need info.]
+
+      :::image type="content" source="../assets/images/teams-toolkit-v2/repo-settings.png" alt-text="Screenshot shows the repo settings.":::
+
+      > [!NOTE]
+      > The `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET` variable must be set as secret.
+      > Utilize the [GitHub environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-variables) to use different sets of variables.
+
+    * `BOT_AZURE_APP_SERVICE_RESOURCE_ID`: Go to the `teamsapp.yml` file. In the `deploy` stage, the values enclosed in `${{}}` are the required variable keys. If you've used the `provision` command from Teams Toolkit, you can locate the values in the environment files in the `.env` folder.
+
+      :::image type="content" source="../assets/images/teams-toolkit-v2/teamsappyml.png" alt-text="Screenshot shows the bot Azure app service resource ID in teamsapp.yml file.":::
 
-### Customize CI workflow
+    * `TEAMS_APP_ID`: Go to the `appPackage/manifest.json` file. Go to `id`, the values enclosed in `${{}}` are the required variable keys. If you've used the `provision` command from Teams Toolkit, you can locate the values in the environment files in the `.env` folder.
 
-To customize the CI workflow, you can do the following:
+      :::image type="content" source="../assets/images/teams-toolkit-v2/manifest.png" alt-text="Screenshot shows the Teams app ID in manifest file.":::
```

Comment:
This image is not shown in review site.
![image](https://github.com/MicrosoftDocs/msteams-docs/assets/25220706/f40bda70-57b9-4b1c-b388-9d21612aadd3)

Reply:
Updated.